src/v/pandaproxy/schema_registry/handlers.cc

```diff
@@ -660,17 +663,37 @@ compatibility_subject_version(server::request_t rq, server::reply_t rp) {
         version = parse_numerical_schema_version(ver).value();
     }
 
-    auto schema = co_await rq.service().schema_store().make_canonical_schema(
-      std::move(unparsed.def));
+    canonical_schema schema;
+    try {
+        schema = co_await rq.service().schema_store().make_canonical_schema(
+          std::move(unparsed.def));
+    } catch (exception& e) {
+        const static std::unordered_set<std::error_code> report{
+          error_code::schema_invalid, error_code::schema_empty};
+        if (is_verbose && report.contains(e.code())) {
```

Comment:
I'm not really a fan of this static, I expect the generated code to be nicer with something like:
```suggestion
        constexpr auto reportable = [](std::error_code ec) {
            constexpr std::array errors{
              error_code::schema_invalid, error_code::schema_empty};
            return absl::c_any_of(
              errors, [ec](error_code e) { return ec == e; });
        };
```

Reply:
Fixed